sunpy/physics/differential_rotation.py

```diff
@@ -579,9 +579,9 @@ def differential_rotate(smap, observer=None, time=None, **diff_rot_kwargs):
         out_meta.pop(key)
 
     # Add a new HGS observer
-    out_meta['hglt_obs'] = new_observer.lat.value
-    out_meta['hgln_obs'] = new_observer.lon.value
-    out_meta['dsun_obs'] = new_observer.radius.to(u.m).value
+    out_meta['hglt_obs'] = new_observer.transform_to('heliographic_stonyhurst').lat.value
```

Comment:
I think it would be better to do the `transform_to` only once first and then extract the values afterwards. For bonus points though you could use https://github.com/sunpy/sunpy/blob/master/sunpy/map/header_helper.py#L193 which I refactored for this purpose.

Reply:
Done. You've reduced my PR to a single line!